src/libs/Navigation/AppNavigator/AuthScreens.js

```diff
@@ -109,11 +118,20 @@ class AuthScreens extends React.Component {
         }, ['meta'], true);
     }
 
-    shouldComponentUpdate(prevProps) {
-        if (prevProps.isSmallScreenWidth !== this.props.isSmallScreenWidth) {
+    shouldComponentUpdate(nextProps) {
+        if (nextProps.isSmallScreenWidth !== this.props.isSmallScreenWidth) {
             return true;
         }
 
+        // Update initialHomeParams only once
+        if (!this.initialHomeParams) {
+            // Either we have a reportID or fetchAllReports resolved with no reports. Otherwise keep waiting
+            if (nextProps.initialReportID || nextProps.initialReportID === '') {
+                this.setInitialHomeParams(nextProps.initialReportID);
+                return true;
+            }
```

Comment:
@marcaaron can you confirm this would work

The idea is:
1. If we have a stored report we'll never end up in here. It'll be the initial value.
2. In case we don't we'll end up here after `fetchAllReports` ends
3. Looking at the code `fetchAllReports` would never fail, but if there's no report the key `CURRENTLY_VIEWED_REPORTID` would be set to `''`

Reply:
I don't see why this wouldn't work, but can also do some testing. I was a little confused on how it works at first, but after looking at the entire PR it makes sense. I have some thoughts on simplifying things.

Reply:
I've left you a note on another thread